src/session/Timer.tsx

```diff
@@ -66,21 +66,16 @@ const Timer = ({
       setIterationCount(1);
 
       if (window.Notification && Notification.permission !== "granted") {
+        const alertMessageByNotificationPermission: {
+          [key in NotificationPermission]: string;
+        } = {
+          granted: "Thanks to accept the notification :)",
+          denied: "You rejected the notification :(Please accept it.",
+          default: "Can not judge to use notification :(Please accept it.",
+        };
```

Comment:
```suggestion
        const alertMessageByNotificationPermission = {
          granted: "Thanks to accept the notification :)",
          denied: "You rejected the notification :(Please accept it.",
          default: "Can not judge to use notification :(Please accept it.",
        } as  const;
```

is simple and robust here, dependent NotificationPermission is enough in fetching key, I think

And I suggest to put this object out of this Timer function, this does not depends any context and variables.

Reply:
I got another idea now. How about this?  
```ts
      if (window.Notification && Notification.permission !== "granted") {
        Notification.requestPermission((result) => {
          const alertMessageByNotificationPermission: {
            [key in typeof result]: string;
          } = {
            granted: "Thanks to accept the notification :)",
            denied: "You rejected the notification :(Please accept it.",
            default: "Can not judge to use notification :(Please accept it.",
          };
          alert(alertMessageByNotificationPermission[result]);
        });
      }
```

My thought of `key in ~`'s advantages are:
- Using `as const` doesn't prevent adding "extra" elements.
- `key in ~` can get good IDE support.

On the other hand, I agree using `as const` is more simpler.
I'd like to hear your opinion again which is better.

Reply:
I can understand your motivation to write key ~.

I have 2 suggestions as below

```typescript
        Notification.requestPermission((result) => {
          alert(
            {
              granted: "Thanks to accept the notification :)",
              denied: "You rejected the notification :(Please accept it.",
              default: "Can not judge to use notification :(Please accept it.",
            }[result]
          );
        });
```

This is simple and typescript supports the branches, my favor.
The constant only use here, less cost, so writing as literal is best, I think.

However if you want to extract to another variable, `as const` makes readonly, I prefer the benefit.

```suggestion
        const alertMessageByNotificationPermission: Readonly<{
          [key in NotificationPermission]: string;
        }> = {
          granted: "Thanks to accept the notification :)",
          denied: "You rejected the notification :(Please accept it.",
          default: "Can not judge to use notification :(Please accept it.",
        };
```

This is a similar way as this.

Reply:
I like it! Let me update.

Reply:
addressed in https://github.com/mobu-of-the-world/mobu/pull/479/commits/2e302af1afd29be45f84b20e793f37ac12db6da8
Cloud you check again?

Reply:
I would update this with https://github.com/mobu-of-the-world/mobu/pull/724